### PR TITLE
[receiver/elasticsearch]: add scraping metrics on cluster level

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
**Description:** 
The receiver now scrapes cluster-level metrics from `/_cluster/stats` endpoint.
There were already some metrics sent with `elasticsearch.cluster.name` resource attribute, but they used different endpoint (`/_cluster/health`). I added the newly emitted metrics to the same resource.

**Link to tracking Issue:** #14635

**Testing:** 
unit, integration

**Documentation:** 
mdatagen

### CODE REVIEW

**Comment by aboguszewski-sumo** [comment]
@djaglowski could this get reviewed soon?

File: receiver/elasticsearchreceiver/testdata/expected_metrics/full.json [hunk 1]
```diff
@@ -2456,6 +2456,57 @@
...
+                     "gauge": {
+                        "dataPoints": [
+                           {
+                              "asInt": "285158912",
+                              "startTimeUnixNano": "1661811689941624000",
+                              "timeUnixNano": "1661811689943245000"
+                           }
+                        ]
+                     },
+                     "name": "jvm.memory.heap.used",
```
Comment (djaglowski) [inline receiver/elasticsearchreceiver/testdata/expected_metrics/full.json:2471]: Why does this appear in this PR? Need a rebase?

Comment (aboguszewski-sumo) [reply receiver/elasticsearchreceiver/testdata/expected_metrics/full.json:2471]: That's one of the metrics that I've added to be emitted in this PR (see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16346/files#diff-9e20fd1298180a1ab0c45e590f71b1e98e893104c458140f5ef744c528f2a960R304-R325).
It is already emitted on node-level, but this is on cluster-level (aggregated by `elasticsearch`). Currently the only way to get this metric is to manually sum by nodes after receiving the data - that's not very convienient.

File: receiver/elasticsearchreceiver/config.go [hunk 2]
```diff
@@ -47,7 +47,11 @@ type Config struct {
 	// See https://www.elastic.co/guide/en/elasticsearch/reference/7.9/cluster.html#cluster-nodes for which selectors may be used here.
 	// If Nodes is empty, no nodes will be scraped.
 	Nodes []string `mapstructure:"nodes"`
-	// SkipClusterMetrics indicates whether cluster level metrics from /_cluster/health should be scraped or not.
+	// ClusterNodes defines the nodes to scrape from /_cluster/stats endpoint.
+	// See https://www.elastic.co/guide/en/elasticsearch/reference/7.9/cluster.html#cluster-nodes for which selectors may be used here.
+	// If ClusterNodes is empty, no nodes will be scraped.
+	ClusterMetricsNodes []string `mapstructure:"cluster_metrics_nodes"`
+	// SkipClusterMetrics indicates whether cluster level metrics from /_cluster/* endpoints should be scraped or not.
```
[the remark this answers is not in the document]
Comment (aboguszewski-sumo) [reply receiver/elasticsearchreceiver/config.go:54]: The set of nodes that already exists (`Nodes`) is for querying the `/_nodes` endpoint. The response contains a JSON with data for every queried node.
The added set of nodes (`ClusterMetricsNodes`) is for querying the `/_cluster/_stats` endpoint. The response contains the data aggregated by every queried node. 
My reasoning was that the user might want to query by a different set of nodes for these two endpoint - for example they might want to query only for some individual nodes, but get aggregated info for all nodes. Maybe it's worth it to remove this possibility for the sake of a simpler interface.

File: receiver/elasticsearchreceiver/testdata/expected_metrics/full.json (hunk 1 above)
Comment (djaglowski) [reply receiver/elasticsearchreceiver/testdata/expected_metrics/full.json:2471]: Makes sense. Thanks for explaining.

File: receiver/elasticsearchreceiver/config.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (aboguszewski-sumo) [reply receiver/elasticsearchreceiver/config.go]: I don't think that it's really something limiting, it even makes sense that someone wants to get data for the same sets of nodes in both cases.
So, I've removed the additional config option and modified information about `Nodes` config in the readme.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**